helm-chart/todoapp/charts/mysql/templates/statefulSet.yml

```diff
@@ -0,0 +1,89 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Release.Name }}-{{ .Chart.Name }}
+  namespace: {{ .Values.common.namespace }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ .Chart.Name }}
+  replicas: {{ .Values.mysql.statefulSet.replicas }}
+  serviceName: mysql
+  template:
+    metadata:
+      labels:
+        app: {{ .Chart.Name }}
+    spec:
+      containers:
+      - name: {{ .Values.mysql.image.name }}
+        image: {{ .Values.mysql.image.repository }}:{{ .Values.mysql.image.version }}
+        env:
+        {{- range $key, $value := .Values.mysql.secret }}
+        - name: {{ $key }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ $.Chart.Name }}-secrets
+              key: {{ $key }}
+        {{- end }}
```

Comment:
Make sure that the `mysql.secret` values in your `values.yaml` file are properly defined. These values should match the keys in the secret and be correctly base64 encoded.